frontend/src/components/NewRunParametersV2.tsx

```diff
@@ -91,13 +107,45 @@ function convertInput(paramStr: string, paramType: ParameterType_ParameterTypeEn
   }
 }
 
+function checkInput(
+  parametersInRealType: any,
+  paramType: ParameterType_ParameterTypeEnum,
+): { validInput: boolean; errorMessage: string } {
+  let isValid: boolean;
+  let eMsg: string;
+  switch (parametersInRealType) {
+    case undefined:
+      isValid = false;
+      eMsg = 'Missing parameter.';
+      break;
+    // TODO(jlyaoyuli): tell the error difference between mismatch type or invalid JSON form.
+    case null:
+      isValid = false;
```

Comment:
Currently, we treat both `non-parsable` and `parsable but invalid type` as invalid type error.